_tools/release_docs/main.go

```diff
@@ -151,6 +164,84 @@ func rewriteCodeBlocks(fileBytes []byte) []byte {
 	return []byte(strings.Join(output, "\n") + "\n")
 }
 
+var (
+	// matches "# headers"
+	atxTitleRE = regexp.MustCompile(`(?m)^\s*#\s+(?P<title>.*)$`)
```

Comment:
`?m` does not affect `.*`, however `.*` already matches across lines (like we do with the `linksRE`). If you look at the doc for [`regexp.Compile`](http://golang.org/pkg/regexp/#Compile), you'll see that it will return the first match that backtracking would find. So we won't span lines with this. If we used [`regexp.CompilePOSIX`](http://golang.org/pkg/regexp/#CompilePOSIX), then we would match across lines.

All `?m` does here is make `^` and `$` also bind to beginning and end of line respectively.